src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -317,7 +317,7 @@ private string GetProcessesToIgnoreKey(long hostHandshake, long clientHandshake,
         private void ValidateRemotePipeSecurityOnWindows(NamedPipeClientStream nodeStream)
         {
             SecurityIdentifier identifier = WindowsIdentity.GetCurrent().Owner;
-#if FEATURE_PIPE_SECURITY
+#if !FEATURE_PIPE_SECURITY
```

Comment:
The actual build error is:
```
BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs(323,34): error CS0122: `System.IO.Pipes.PipeSecurity.PipeSecurity(System.Runtime.InteropServices.SafeHandle, System.Security.AccessControl.AccessControlSections)' is inaccessible due to its protection level [/Users/ankit/dev/msbuild/src/XMakeBuildEngine
/Microsoft.Build.csproj]
```

Yeah, we do seem to have the classes. `FEATURE_PIPE_SECURITY` is disabled for Mono builds in `dir.props`. If I enable that, then I hit:

```
../Shared/NodeEndpointOutOfProcBase.cs(209,94): error CS0103: The name `AccessControlType' does not exist in the current context [/Users/ankit/dev/msbuild/src/XMakeBuildEngine/Microsoft.Build.csproj]
```

.. which seems to be because `using System.Security.AccessControl` is guarded by `#if FEATURE_SECURITY_PERMISSIONS` . Changing that to `#if FEATURE_SECURITY_PERMISSIONS || FEATURE_PIPE_SECURITY` (maybe only `FEATURE_PIPE_SECURITY` required here?), the build succeeds with all the tests passing.